### PR TITLE
feat - WIP229 - Edit questions

### PR DESCRIPTION
**Pull request by ramonrf**
#### What does this PR do?
- A new modal was added to edit questions.
- Edit Question Modal was moved to a new component.
- Answer Question Modal was moved to a new component.
#### What are the relevant tickets?
WIP-229 Ability to edit questions

### CODE REVIEW

File: src/components/SubmitAnswerModal.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+          <Modal.Title>Question #{question.question_id}: {question.question}</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <form onSubmit={this.submitAnswer}>
+            <textarea
+              type="text"
+              className="form-control"
+              rows="5"
+              placeholder="Insert answer here..."
+              ref={(thisElement) => { this.answerInput = thisElement; }}
```
Comment (rominavarela) [inline src/components/SubmitAnswerModal.js]: I guess this `ref` is for removing the `onChange` listener. Does this stops working if we put it back the `value` property? Refs are kind of dirty magic cards

File: src/components/EditQuestionModal.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { putQuestion, fireAlert } from '../actions';
```
Comment (alan-maldonado) [inline src/components/EditQuestionModal.js]: Nice 👍

File: src/components/EditQuestionModal.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    fireAlert: PropTypes.func.isRequired,
+    putQuestion: PropTypes.func.isRequired,
+    onClose: PropTypes.func.isRequired,
+    onSubmitSuccess: PropTypes.func.isRequired,
+  }
+
+  submitQuestionEdit = (e) => {
+    const newValue = this.editInput.value;
+    e.preventDefault();
+    if (newValue === '') {
```
Comment (alan-maldonado) [inline src/components/EditQuestionModal.js]: you could change this condition to `if (!newValue) { }`, but it's ok.

File: src/components/EditQuestionModal.test.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import Modal from './EditQuestionModal';
+import { mount } from 'enzyme';
+import storeFake from '../stores';
+import { Provider } from 'react-redux';
+
```
Comment (alan-maldonado) [inline src/components/EditQuestionModal.test.js]: Please move libraries to top, and project files with relatives path to bottom.

File: src/components/ListQuestions.js [hunk 5]
```diff
@@ -54,50 +53,47 @@ class List extends Component {
...
-      }
-      this.setState(resetModal);
-      this.props.fireAlert('success', 'Success', 'Answer successfully saved');
-    });
+    this.setState(resetModal);
+    this.props.fetchQuestions();
   }
 
-/* eslint class-methods-use-this: ["error", { "exceptMethods": ["submitVote"] }] */
+  /* eslint class-methods-use-this: ["error", { "exceptMethods": ["submitVote"] }] */
```
Comment (alan-maldonado) [inline src/components/ListQuestions.js]: why is this needed?

File: src/components/SubmitAnswerModal.js (hunk 1 above)
Comment (alan-maldonado) [inline src/components/SubmitAnswerModal.js]: I really like somethings use refs, moreover when I want an uncontrolled component. Not always is needed to keep track o each change, I think it is good because we only need the value on submit.

File: src/components/ListQuestions.js (hunk 5 above)
Comment (ramonrf) [inline src/components/ListQuestions.js]: That line was already there, it seems that it was not aligned with the code.

File: src/components/EditQuestionModal.js (hunk 3 above)
Comment (codinronan) [inline src/components/EditQuestionModal.js]: Actually I think that is preferable. This case will miss null and undefined.

File: src/components/ListQuestions.js (hunk 5 above)
Comment (codinronan) [inline src/components/ListQuestions.js]: ESLint will tell you to not make a function a class method if it does not use `this`. If it doesn't use `this`, it doesn't need to be a class method right?

So this eslint directive overrides that error.

File: src/components/SubmitAnswerModal.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    fireAlert: PropTypes.func.isRequired,
+    answerQuestion: PropTypes.func.isRequired,
+    onClose: PropTypes.func.isRequired,
+    onSubmitSuccess: PropTypes.func.isRequired,
+  }
+
+  submitAnswer = (e) => {
+    e.preventDefault();
+    const answerValue = this.answerInput.value;
+    if (answerValue === '') {
```
Comment (codinronan) [inline src/components/SubmitAnswerModal.js]: Again, `if(!answerValue) { ... }`

File: src/components/SubmitAnswerModal.js (hunk 1 above)
Comment (codinronan) [inline src/components/SubmitAnswerModal.js]: I agree with @alan-maldonado :)

File: src/components/SubmitAnswerModal.js (hunk 6 above)
Comment (alan-maldonado) [inline src/components/SubmitAnswerModal.js]: ➕
